### ISSUE TITLE
`intl` date handler shows wrong language when default and user language is different in panel

### ISSUE THREAD

**Issue by afbora** [k-next/kirby#5913]
`intl` date handler shows wrong language when default and user language is different in panel. I see all UI strings/variables as TR (user language is TR, EN default language) but `intl` date handler shows English. 

For ex: following info output is `04 Nov, 2023`, but it should be `04 Kas, 2023`

````yaml
type: pages
info: >
  {{ model.createdAt.toDate('dd MMM, YYYY') }}
````

### config.file

````php
<?php

return [
    'languages' => true,
    'languages.detect' => true,
    'panel' => [
        'language' => 'tr',
    ],
    'date' => [
        'handler' => 'intl'
    ],
];
````

### Language

Multi-language enabled and default language is EN and locale `en_US`

### User

Logged user language is Turkish.

### Kirby 
3.9.7

**Comment by distantnative** [comment]
So Panel user language is Turkish but you're editing English content or Turkish content?

**Comment by afbora** [comment]
Yes, panel user language TR and editing English content. All panel UI and language dropdown (fetching query from collections) options TR except date.

**Comment by distantnative** [comment]
This sounds similar to https://github.com/getkirby/kirby/issues/5269.

Where there's no obvious answer if the Panel should request the backend passing the content language or the user language as current language.

In your example, I could also see people arguing the info text should match the current content language, not the UI language.
